src/freenet/clients/http/WelcomeToadlet.java

```diff
@@ -110,6 +110,45 @@ public boolean allowPOSTWithoutPassword() {
     	return true;
     }
 
+    public boolean showSearchBox()  {
+        // Only show it if Library is loaded.
+        if (node.pluginManager != null &&
+            node.pluginManager.isPluginLoaded("plugins.Library.Main")) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public void addSearchBox(HTMLNode contentNode)  {
+        // This function still contains legacy cruft because we might
+        // need that again when Library becomes usable again.
+        HTMLNode searchBox = contentNode.addChild("div", "class", "infobox infobox-normal");
+        searchBox.addAttribute("id", "search-freenet");
+        searchBox.addChild("div", "class", "infobox-header").addChild("span", "class", "search-title-label", NodeL10n.getBase().getString("WelcomeToadlet.searchBoxLabel"));
+        HTMLNode searchBoxContent = searchBox.addChild("div", "class", "infobox-content");
+        // Search form
+        if(node.pluginManager != null &&
```

Comment:
if I go for that, I should additionally create a “showSearchBoxLoading” function.